x-pack/plugins/cross_cluster_replication/server/routes/api/auto_follow_pattern.test.js

```diff
@@ -125,11 +144,51 @@ describe('[CCR API Routes] Auto Follow Pattern', () => {
 
     it('should return a single resource even though ES return an array with 1 item', async () => {
       const autoFollowPattern = getAutoFollowPatternMock();
-      setHttpRequestResponse(null, { patterns: [autoFollowPattern] });
+      const esResponse = { patterns: [autoFollowPattern] };
+
+      setHttpRequestResponse(null, esResponse);
 
       const response = await routeHandler({ params: { id: 1 } });
       expect(Object.keys(response)).toEqual(DESERIALIZED_KEYS);
     });
+  });
+
+  describe('delete()', () => {
+    beforeEach(() => {
+      resetHttpRequestResponses();
+      routeHandler = routeHandlers.delete;
+    });
 
+    it('should delete a single item', async () => {
+      setHttpRequestResponse(null, { acknowledge: true });
+
+      const response = await routeHandler({ params: { id: 'a' } });
+
+      expect(response.itemsDeleted).toEqual(['a']);
+      expect(response.errors).toEqual([]);
+    });
+
+    it('should accept a list of ids to delete', async () => {
+      setHttpRequestResponse(null, { acknowledge: true });
+      setHttpRequestResponse(null, { acknowledge: true });
+      setHttpRequestResponse(null, { acknowledge: true });
+
+      const response = await routeHandler({ params: { id: 'a,b,c' } });
+
+      expect(response.itemsDeleted).toEqual(['a', 'b', 'c']);
+    });
+
+    it('should catch error and return them in array', async () => {
```

Comment:
Can we add another test to ensure a single failure doesn't cause the endpoint to return early?

```js
    it('should continue deleting items after an error', async () => {
      const error = new Error('something went wrong');
      error.response = '{ "error": {} }';

      setHttpRequestResponse(null, { acknowledge: true });
      setHttpRequestResponse(error);
      setHttpRequestResponse(null, { acknowledge: true });

      const response = await routeHandler({ params: { id: 'a,b,c' } });

      expect(response.itemsDeleted).toEqual(['a', 'c']);
    });
```

Reply:
Not sure I understood you 😊 Isn't this test enough to test that?

Reply:
I might be overthinking things, but from looking at your test I wasn't sure if subsequent requests which resolve _after_ the error would still be part of the response. I was just hoping to make that clearer.

Reply:
Ok, let me see how it could be clearer. I know it does as I know the implementation (loops over 'a', 'b', 'c' and returns 'a', 'c' even when 'b' failed) but you're right it readability could be improved.